lib/krb5/cache.c

```diff
@@ -443,10 +443,7 @@ environment_changed(krb5_context context)
 	 strncmp(context->default_cc_name, "API:", 4) == 0))
 	return 1;
 
-    if(issuid())
-	return 0;
-
-    e = getenv("KRB5CCNAME");
+    e = secure_getenv("KRB5CCNAME");
```

Comment:
Yes, actually.  I pondered that for a bit myself.  `context->default_cc_name_env` basically gets the value of `secure_getenv("KRB5CCNAME")`, so if `issuid()` is true then `secure_getenv("KRB5CCNAME")` is `NULL` and `context->default_cc_name_env` is too, so we still `return 0` here.